packages/pocket-ic/src/common/rest.rs

```diff
@@ -723,3 +723,135 @@ impl Topology {
             .map(|(id, _)| *id)
     }
 }
+
+#[derive(
+    Clone, Serialize, Deserialize, Debug, Hash, Eq, PartialEq, Ord, PartialOrd, JsonSchema,
+)]
+pub enum CanisterHttpMethod {
+    GET,
+    POST,
+    HEAD,
+}
+
+#[derive(
+    Clone, Serialize, Deserialize, Debug, Hash, Eq, PartialEq, Ord, PartialOrd, JsonSchema,
+)]
+pub struct CanisterHttpHeader {
+    pub name: String,
+    pub value: String,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, JsonSchema)]
+pub struct RawCanisterHttpRequest {
+    pub subnet_id: RawSubnetId,
+    pub request_id: u64,
+    pub http_method: CanisterHttpMethod,
+    pub url: String,
+    pub headers: Vec<CanisterHttpHeader>,
+    pub body: Option<Vec<u8>>,
```

Comment:
In the raw types defined in this file, we have at least so far used a slightly more efficient representation of Vec<u8>s, by using:
```
#[serde(deserialize_with = "base64::deserialize")]
#[serde(serialize_with = "base64::serialize")]
```

Reply:
Does this also work with `Option`? Otherwise, I should be able to remove the Option if needed.

Reply:
Yes, by providing a `deserialize_with = my_function`, as shown [here](https://users.rust-lang.org/t/solved-serde-deserialize-with-for-option-s/12749/4).

Reply:
>if needed.

This is merely an optimization and uniformity thing.

Reply:
> Yes, by providing a `deserialize_with = my_function`

I don't want to implement a custom deserializer for this - so I switched from `Option<Vec<u8>>` to `Vec<u8>`.

Reply:
Ok - If you think that it is valid to represent None as [] in all cases. Otherwise it would also be fine to ignore the de/serialization optimization and use the original type.

Reply:
> If you think that it is valid to represent None as [] in all cases

I'm actually not sure why the body is optional in the first place (maybe so that you can pass `None` for GET requests that typically don't have any body) so I think it is fine to have `body: Vec<u8>`. There are also the following arguments in favor of `body: Vec<u8>`:
- the body is [non-optional](https://github.com/dfinity/ic/blob/65c3775ebcd52c5c887efb9c27a442b38c9e5537/rs/https_outcalls/service/proto/canister_http_service/v1/proto.proto#L19) in the type used in the production component making the actual HTTPS outcall
- the response body is non-optional, too, so we're consistent here

The only argument in favor of `body: Option<Vec<u8>>` is:
- consistency with the management canister type where the body has such a type (but we don't simply copy the management canister type anyway as we add a few more non-optional fields)